chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/state/RoomUpdated.kt

```diff
@@ -0,0 +1,33 @@
+package com.pusher.chatkit.rooms.state
+
+import com.pusher.chatkit.state.ChatkitState
+import org.reduxkotlin.reducerForActionType
+
+internal data class RoomUpdated(
+    val room: JoinedRoomInternalType,
+    val unreadCount: Int
+)
+
+internal val roomUpdatedReducer =
+    reducerForActionType<ChatkitState, RoomUpdated> { state, action ->
+
+            var joinedRoomsState = JoinedRoomsState(
+                    hashMapOf(
+                            Pair(action.room.id, action.room)),
+                    hashMapOf(
+                            Pair(action.room.id, action.unreadCount))
+            )
+
+            if (state.joinedRoomsState != null) {
+                    joinedRoomsState = JoinedRoomsState(
+                            state.joinedRoomsState.rooms
+                                    .filterNot { it.key == action.room.id }
+                                    .plus(Pair(action.room.id, action.room)),
```

Comment:
to make it concise and Kotlin-idiomatic:
```suggestion
                                    .plus(action.room.id to action.room),
```

Reply:
👍